engine/src/main/java/org/terasology/logic/players/CameraClientSystem.java

```diff
@@ -43,19 +44,27 @@
     EntityManager entityManager;
 
     @ReceiveEvent
-    public void onAutoMountCamera(OnChangedComponent event, EntityRef client, AutoMountCameraComponent autoMountCameraComponent, ClientComponent clientComponent) {
-        if (clientComponent.camera.exists()) {
-            client.send(new ResetCameraEvent());
+    public void ensureCameraEntityCreatedOnChangedClientComponent(OnChangedComponent event, EntityRef client, AutoMountCameraComponent autoMountCameraComponent, ClientComponent clientComponent) {
+        if (localPlayer.getClientEntity().equals(client)) {
+            ensureCameraEntityCreated();
         }
     }
 
```

Comment:
Agreed.  Ill add a class description.  I was hoping the method names would be enough.  :)